test/jdk/java/nio/channels/unixdomain/Bind.java

```diff
@@ -0,0 +1,331 @@
+    static ServerSocketChannel server;
+    static SocketChannel client, accept1;
+
+    public static void main(String args[]) throws Exception {
+        if (!supported()) {
+            System.out.println("Unix domain channels not supported");
+            return;
+        }
+        spath = Path.of("server.sock");
+        cpath = Path.of("client.sock");
+        sAddr = UnixDomainSocketAddress.of(spath);
+        cAddr = UnixDomainSocketAddress.of(cpath);
+        nullAddr = UnixDomainSocketAddress.of("");
+        UNNAMED = nullAddr;
+        runTests();
+    }
+
+    static boolean supported() {
+        try {
+            SocketChannel.open(StandardProtocolFamily.UNIX).close();
+        } catch (UnsupportedOperationException e) {
+            return false;
+        } catch (Exception e) {
+            return true; // continue test to see what problem is
+        }
+        return true;
+    }
+
+    static interface ThrowingRunnable {
+        public void run() throws Exception;
+    }
+
+    static void init() throws IOException {
+        Files.deleteIfExists(cpath);
+        Files.deleteIfExists(spath);
+        client = null; server = null; accept1 = null;
+    }
+
+    static void checkNormal(ThrowingRunnable r) {
+        try {
+            init();
+            r.run();
+            System.out.println("PASS:");
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        } finally {
+            cleanup();
+        }
+    }
+
+    static void checkException(Class<? extends Exception> expected, ThrowingRunnable r) {
+        try {
+            init();
+            r.run();
+            throw new RuntimeException("Exception expected");
+        } catch (Exception e) {
+            if (!expected.isAssignableFrom(e.getClass())) {
+                String msg = "Expected: " + expected + " Got: " + e.getClass();
+                throw new RuntimeException(msg);
+            }
+            System.out.println("PASS: Got " + e);
+        } finally {
+            cleanup();
+        }
+    }
+
+    static void cleanup() {
+        try {
+            if (server != null)
+                server.close();
+            if (client != null)
+                client.close();
+            if (accept1 != null)
+                accept1.close();
+        } catch (IOException e) {}
+    }
+
+    static void assertClientAddress(SocketAddress a) {
+        assertAddress(a, cAddr, "client");
+    }
+
+    static void assertServerAddress(SocketAddress a) {
+        assertAddress(a, sAddr, "server");
+    }
+
+    static void assertAddress(SocketAddress a, UnixDomainSocketAddress a1, String s) {
+        if (!(a instanceof UnixDomainSocketAddress)) {
+            throw new RuntimeException("wrong address type");
+        }
+        UnixDomainSocketAddress ua = (UnixDomainSocketAddress)a;
+        if (!a.equals(a1))
+            throw new RuntimeException("this is not the " + s + " address");
```

Comment:
Could you consider turning this test into a `testng` test so that asssertEquals can be used?
Alternatively if you want to retain the ability to run the test as a standalone java program out of the repository then consider modifying the assertXxxx methods so that the exception message prints both parts: `<what>` is not equal to `<what>`